mava/utils/jax.py

```diff
@@ -42,3 +47,18 @@ def merge_leading_dims(x: chex.Array, num_dims: chex.Numeric) -> chex.Array:
 
     new_shape = (np.prod(x.shape[:num_dims]),) + x.shape[num_dims:]
     return x.reshape(new_shape)
+
+
+def unreplicate_learner_state(
+    learner_state: Union[LearnerState, RNNLearnerState]
+) -> Union[LearnerState, RNNLearnerState]:
+    """Unreplicates a learner state.
+
+    This function takes a learner state, and removes the axes associated with device replication
+    and the `update batch size`. The unreplication process is essential to store the learner state
+    in a checkpoint.
+
+    Note:
+        The function internally uses `jax_utils.unreplicate` twice to remove the necessary axes.
+    """
+    return jax_utils.unreplicate(jax_utils.unreplicate(learner_state))  # type: ignore
```

Comment:
```suggestion
def unreplicate_learner_state(
    learner_state: Union[LearnerState, RNNLearnerState],
    unreplicate_depth: int = 2,
) -> Union[LearnerState, RNNLearnerState]:
    """Unreplicates a learner state.

    This function takes a learner state and removes some number of axes, associated with parameter 
    duplication. This is typically one axis for device replication, and one for the `update batch size`.
    """
    return jax.tree_map(lambda x: x[(0,) * unreplicate_depth], learner_state)
```

Reply:
I couldn't commit suggestion as the code was outdated so I tried to make co-author commit but it didn't work as it should :sweat_smile: https://github.com/instadeepai/Mava/pull/986/commits/5d426dc259318077a24f88ad76bdcac606b873a8